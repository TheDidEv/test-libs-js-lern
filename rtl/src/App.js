import { useEffect, useState } from "react";

function App() {
  const [data, setData] = useState(null);
  const [toggle, setToggle] = useState(false);

  const onClick = () => setToggle(prev => !prev);

  useEffect(() => {
    setTimeout(() => {
      setData({})
    }, 100);
  })

  return (
    <div>
      {toggle === true && <div>toggle</div>}
      {data && <div style={{ color: 'red' }}>data</div>}
      <h1>Hello world</h1>
      <button onClick={onClick}>bttn</button>
      <input type="text" placeholder="input value" />
    </div>
  );
}

export default App;
