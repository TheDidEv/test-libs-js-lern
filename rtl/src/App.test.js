import { render, screen } from '@testing-library/react';
import App from './App';


describe('TESTAPP', () => {
  test('renders learn react link', () => {
    render(<App />);
    const hellowWorldElem = screen.getByText(/Hello world/i);// //...//i - регулярний вираз який означає що ми ігноруємо регістр(<h1>, <div>, ...)
    expect(hellowWorldElem).toBeInTheDocument();//toBeInTheDocument - перевіряє чи з'явилось в dom дереві

    const bttnElem = screen.getByRole('button');//getByRole - перевіряє чи є такий елемент
    expect(bttnElem).toMatchSnapshot();//toMatchSnapshot - робить snapshot

    // const inputElem = screen.getByPlaceholderText(/input value/i);
    // expect(inputElem) = screen.toBeInTheDocument();

    // screen.debug();//повертає розмітку
  });

  test('renders learn react link2', async () => {
    render(<App />);
    //findBy/all - знаходить елемент. Повертає обєкт завернутий в Promise. Використовується для роботи з асинхроним кодом.
    //getBy/all - обовязково повинен знайти який елемент, якщо елемен не знаходиться видється помилка і тест падає. Повертає обєкт
    //qyeryBy/all - ми можемо переконатися що якогось елементу немає.

    // const hellowWorldElem = screen.queryByText(/Helo/i);
    // expect(hellowWorldElem).toBeNull();//якщо значення тесту повертає null то тест пройдено.

    //screen.debug();
    const hellowWorldElem = await screen.findByText(/data/i);//працює з asinc await
    expect(hellowWorldElem).toBeInTheDocument();
    expect(hellowWorldElem).toHaveStyle({ color: 'red' });
    //screen.debug();
  });
});

