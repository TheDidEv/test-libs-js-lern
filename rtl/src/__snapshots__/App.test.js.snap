// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTAPP renders learn react link 1`] = `
<button>
  bttn
</button>
`;

exports[`renders learn react link 1`] = `
<button>
  bttn
</button>
`;
